docs/verifier/on-chain-verification/cross-chain.md

```diff
@@ -0,0 +1,213 @@
+
+We override the `_update` function to use the `beforeTokenTransfer` modifier, ensuring that every time a user calls the `mint` function, it will trigger this check. The contract verifies that the user has submitted the necessary proofs and is verified before allowing token minting. Once the tokens are minted, the `mint` function sets the `isClaimed` status to `true` to prevent users from reclaiming the airdrop.
+
+In summary, when a user calls the `mint` function, the smart contract checks if the user has been verified (submitted proofs and got verified by Universal Verifier). If verified, it mints 5 tokens for the user.
+
+**Note:** You can observe that you don't need to implement the Universal Verifier smart contract yourself. Instead, you only need the address of the deployed Universal Verifier smart contract to use in your custom smart contract implementation.
+
+#### B. Deploy the Smart Contract:
+
+```javascript
+const { ethers } = require("hardhat");
+
+async function main() {
+	const universalVerifierAddress = "0xfcc86A79fCb057A8e55C6B853dff9479C3cf607c"; 
+	const verifierName = "ZKAirdropVerifier";
+	const verifierSymbol = "zkERC20";
+
+	const verifier = await ethers.deployContract(verifierName, [
+		universalVerifierAddress,
+		verifierName,
+		verifierSymbol,
+	]);
+	await verifier.waitForDeployment();
+	console.log(verifierName, " contract address:", await verifier.getAddress());
+}
+
+main()
+	.then(() => process.exit(0))
+	.catch((error) => {
+		console.error(error);
+		process.exit(1);
+	});
+```
+deploy the smart contracts using the following command:
+
+`npx hardhat run scripts/deploy.js --network amoy`
+
+ Ensure that the smart contract address is saved to interact with the DApp.
+### Step 3: User Claiming The Airdrop
+
+#### A. Obtaining and Signing the Verifiable Credential
+
+Users need to obtain a `POH Credential`, which proves their Proof of Humanity (PoH). This credential can be issued by a trusted identity provider on the Issuer Chain.
+
+On the Privado ID issuer node, credentials will be issued using the Privado chain, which acts like a base layer for issuing identities.
+
+**Note:** Follow this tutorial to learn how to get a Proof of Humanity (PoH) credential from the Privado ID demo issuer.
+
+#### B. User Submitting Proof to Verification Chain
+
+The user submits the ZK proof and signed data to the Universal Verifier smart contract on the verification chain. For this example, we are using Polygon Amoy to submit proof.
+
+The Universal Verifier smart contract verifies the ZK proof and checks the signed GIST Root and issuer state data.
+
+Once the user submits the proof and gets verified by the Universal Verifier, they can claim tokens by calling the `mint` function in the `ZKAirdropVerifier` smart contract.
+
+Visit the [Set ZKP Request](/docs/verifier/on-chain-verification/set-zkp-request.md) section to learn how users can submit their proofs. This request can be reused for all users for a specific query.
```

Comment:
In this chapter you speak about submitting proof, however only in the last paragraph refer to set ZKP Request section, which is something that should precede the the submit.
I would suggest to segregate the last paragraph to a separate chapter and put it **before** the submit proof chapter.